src/utils/PolygonConnector.h

```diff
@@ -106,51 +195,285 @@ class PolygonConnector
+     * \param vertex The vertex to add.
+     */
+    void addVertex(ExtrusionLine& polygonal, const ExtrusionJunction& vertex) const;
+
+    /*!
+     * Tests whether this is a closed polygonal object, rather than a polyline.
+     *
+     * Polygons are always closed, so this overload will always return true.
+     * \param polygonal The polygonal object to check.
+     * \return ``true``, indicating that this polygon is closed.
+     */
+    bool isClosed(Polygon& polygonal) const;
+
+    /*!
+     * Tests whether this is a closed polygonal object, rather than a polyline.
+     *
+     * If the endpoints of the extrusion line meet, it is a closed shape. If
+     * not, it is open.
+     * \param polygonal The polygonal object to check.
+     * \return ``true`` if that shape is closed, or ``false`` otherwise.
+     */
+    bool isClosed(ExtrusionLine& polygonal) const;
+
+    /*!
+     * Get the amount of space in between the polygons at the given connection.
+     *
+     * The space is the length of the connection, minus the width of the
+     * lines at the two endpoints.
+     * \param connection The connection to calculate the space of.
+     */
+    template<typename Polygonal>
+    coord_t getSpace(const PolygonConnection<Polygonal>& connection) const
+    {
+        const coord_t from_width = interpolateWidth(connection.from_point, (*connection.from_poly)[connection.from_segment], (*connection.from_poly)[(connection.from_segment + 1) % connection.from_poly->size()]);
+        const coord_t to_width = interpolateWidth(connection.to_point, (*connection.to_poly)[connection.to_segment], (*connection.to_poly)[(connection.to_segment + 1) % connection.to_poly->size()]);
+        return vSize(connection.to_point - connection.from_point) - from_width / 2 - to_width / 2;
+    }
+
+    /*!
+     * Get the local width at a certain position along a line segment.
+     *
+     * If the line segment has a variable width, the local line width will be
+     * interpolated between the two endpoints.
+     * \param position The position at which to get the line width. This should
+     * be (approximately) in between the position of the two vertices.
+     * \param a One of the vertices between which to interpolate.
+     * \param b The other vertex between which to interpolate.
+     */
+    template<typename Vertex>
+    coord_t interpolateWidth(const Point position, Vertex a, Vertex b) const
+    {
+        const coord_t total_length = vSize(getPosition(a) - getPosition(b));
+        if(total_length == 0) //Prevent division by 0 when the vertices are on top of each other.
+        {
+            return getWidth(a); //Just return one of them. They are on top of each other anyway.
+        }
+        const coord_t position_along_length = vSize(position - getPosition(a));
+        return getWidth(b) * position_along_length / total_length + getWidth(a) * (total_length - position_along_length) / total_length;
+    }
```

Comment:
I tried this with the following implementation:

```
Point a_pos = getPosition(a);
        Point b_pos = getPosition(b);
        if(a_pos == b_pos) //Prevent division by 0 when the vertices are on top of each other.
        {
            return getWidth(a); //Just return one of them. They are on top of each other anyway.
        }
        const Point delta_position = position - a_pos;
        const Point delta_segment = b_pos - a_pos;
        const coord_t numerator = (std::abs(delta_segment.X) > std::abs(delta_segment.Y)) ? delta_position.X : delta_position.Y;
        const coord_t denominator = (std::abs(delta_segment.X) > std::abs(delta_segment.Y)) ? delta_segment.X : delta_segment.Y;
        return round_divide(numerator * getWidth(b), denominator) + round_divide((denominator - numerator) * getWidth(a), denominator);
```

Without the `std::abs` it resulted in a division by zero, since `delta_segment` can also be negative (it had a vector of `[-1000, 0]`). With them, it causes the tests to fail because it can't create a bridge somewhere.
So I think I'll leave it like this. It's probably not as efficient as it could be, but micro-optimising can be done if performance becomes a problem.

Reply:
Yes, `abs` with integers is a single instruction iirc. But even with `abs` the result was significantly off.